### PR TITLE
added org_policy resource updated file

### PR DESCRIPTION
**Pull request by DeepakMA07**
Fixes [google_org_policy_policy resource](https://github.com/hashicorp/terraform-provider-google/issues/12363)

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

``` In resource "google_org_policy_policy"   to resolve the [permadiff](https://github.com/hashicorp/terraform-provider-google/issues/12363), we have changed data-type of filed "rules" from "TypeList" to "TypeSet"  in file resource_org_policy_policy.go```

### CODE REVIEW

**Review by hao-nan-li — COMMENTED** [review]
Looks like this PR is a draft, I just want to double-check if it's still WIP or are you looking for a review?

**Comment by DeepakMA07** [comment]
> Looks like this PR is a draft, I just want to double-check if it's still WIP or are you looking for a review?

Hi @hao-nan-li, yahhh at the starting I forgot to put it as in draft, we are working on it once done I'll move it to ready to be review.

**Comment by hao-nan-li** [comment]
> > Looks like this PR is a draft, I just want to double-check if it's still WIP or are you looking for a review?
> 
> Hi @hao-nan-li, yahhh at the starting I forgot to put it as in draft, we are working on it once done I'll move it to ready to be review.

Sounds good!

**Comment by DeepakMA07** [comment]
```release-note:bug
org_policy_policy: fixed perma-diff in `resource_org_policy_policy`
```

File: mmv1/third_party/terraform/tests/resource_org_policy_policy.go [hunk 1]
```diff
@@ -0,0 +1,535 @@
...
+
+	dcl "github.com/GoogleCloudPlatform/declarative-resource-client-library/dcl"
+	orgpolicy "github.com/GoogleCloudPlatform/declarative-resource-client-library/services/google/orgpolicy"
+
+	"github.com/hashicorp/terraform-provider-google/google/tpgdclresource"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+func ResourceOrgPolicyPolicy() *schema.Resource {
```
Comment (hao-nan-li) [inline mmv1/third_party/terraform/tests/resource_org_policy_policy.go:37]: Is there any particular reason you added the fields in a handwritten manner (meaning not adding them in OrgPolicy.yaml)? If you add them in OrgPolicy.yaml, this file should be generated automatically.

**Comment by hao-nan-li** [comment]
Could you also sign for CLA, and also update the changelog according to the release note guide?

File: mmv1/third_party/terraform/tests/resource_org_policy_policy.go (hunk 1 above)
Comment (DeepakMA07) [reply mmv1/third_party/terraform/tests/resource_org_policy_policy.go:37]: Hi @hao-nan-li, The reason behind adding this file is that repository does not contains "resource_org_policy_policy.go" file nor the OrgPolicy.yaml  .

Our actual solution/modification is in the "resource_org_policy_policy.go" file that's why we added the file here.

**Comment by DeepakMA07** [comment]
> Could you also sign for CLA, and also update the changelog according to the release note guide?

We already sign the CLA, att
image (38)](https://github.com/GoogleCloudPlatform/magic-modules/assets/82449864/86715c20-bd09-4d63-b5de-71e688bd7924)
GoogleCloudPlatform/magic-modules/assets/82449864/0fa034eb-14ec-4f4a-a27f-86de8f545afe)
aching the snap for your refence also we update the release note.

Let us know if something is missing.

File: mmv1/third_party/terraform/tests/resource_org_policy_policy.go (hunk 1 above)
Comment (hao-nan-li) [reply mmv1/third_party/terraform/tests/resource_org_policy_policy.go:37]: resource_org_policy_policy.go should be generated to Terraform-google-provider, you should not add  `resource_org_policy_policy.go`. 
https://googlecloudplatform.github.io/magic-modules/get-started/generate-providers/

**Pull request closed without merging**